x-pack/solutions/chat/plugins/wci-index-source/server/routes/configuration.ts

```diff
@@ -61,4 +64,46 @@ export const registerConfigurationRoutes = ({ router, core, logger }: RouteDepen
       }
     }
   );
+
+  router.get(
+    {
+      path: '/internal/wci-index-source/indices-autocomplete',
+      security: {
+        authz: {
+          requiredPrivileges: [apiCapabilities.manageWorkchat],
+        },
+      },
+      validate: {
+        query: schema.object({
+          index: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    async (ctx, request, res) => {
+      try {
+        const { elasticsearch } = await ctx.core;
+        const pattern = request.query.index || '';
```

Comment:
CAT'ing on `*` (or anything lesser than 3 chars length) is dangerous for perfs on large clusters, I would throw if `query.length < 3` (and adapt the fetching logic in the UI accordingly to not fire if the query is less than 3 chars)

(this is how it's done on other parts of Kibana fetching data views, aliases or indices afaik).

Reply:
I have a couple more questions:

1. `initialData` doesn't work - I still get a network call for empty string weirdly, could it be because of my debounce implementation?
2. So if the user enters just 2 letters, then we return no results to the user? But UX might be weird in this case, can we somehow tell the user to continue typing - in the error message, popup or anything, or we just leave it as is?

See demo: 

https://github.com/user-attachments/assets/cfe444df-ed38-4b45-ae32-9d47b36d943e